Dockerfile

```diff
@@ -1,21 +1,57 @@
-FROM ubuntu:bionic
+# syntax=docker/dockerfile:experimental
+
+# Build stage: Install python dependencies
+# ===
+FROM ubuntu:focal AS python-dependencies
+RUN apt-get update && apt-get install --no-install-recommends --yes python3-pip python3-setuptools
+ADD requirements.txt /tmp/requirements.txt
+RUN --mount=type=cache,target=/root/.cache/pip pip3 install --user --requirement /tmp/requirements.txt
+
+
+# Build stage: Install yarn dependencies
+# ===
+FROM node:12-slim AS yarn-dependencies
+WORKDIR /srv
+ADD package.json .
+RUN --mount=type=cache,target=/usr/local/share/.cache/yarn yarn install
+
+# Build stage: Run "yarn run build-js"
+# ===
+FROM yarn-dependencies AS build-js
+ADD static/js static/js
+ADD webpack.config.js .
+RUN yarn run build-js
+
+# Build stage: Run "yarn run build-css"
+# ===
+FROM yarn-dependencies AS build-css
+ADD static/sass static/sass
+RUN yarn run build-css
+
+# Build the production image
+# ===
+FROM ubuntu:focal
+
+ADD . .
+# Install python and import python dependencies
+RUN apt-get update && apt-get install --no-install-recommends --yes python3-lib2to3 python3-pkg-resources ca-certificates libsodium-dev
+COPY --from=python-dependencies /root/.local/lib/python3.8/site-packages /root/.local/lib/python3.8/site-packages
+COPY --from=python-dependencies /root/.local/bin /root/.local/bin
```

Comment:
Sorry I'll make you change once again. @nottrobin proposed a solution on maas.io to remove the 'hard" dependency that we have on minors versions of python: https://github.com/canonical-web-and-design/maas.io/pull/450/files#diff-3254677a7917c6c01f55212f86c57fbfR39-R49

Would you mind making the upgrade here too?

Reply:
@nottrobin not changing it. @tbille will give an explanation.